metabase/metabase
# add urls for dashboard tabs

Part of epic https://github.com/metabase/metabase/issues/29502

### Description

This PR adds distinct URLs for dashboard tabs. 

When a tab is selected, its slug should appear in the URL, except when there is only 1 or no tabs on the dashboard.

When adding a new tab, the slug should not appear until the tab is saved.

When entering the URL with the slug into the browser, the tab referenced in the slug should be selected upon loading the dashboard. If the slug is invalid (tab id not found) the first tab should be selected and the slug should be replaced.

The URLs should be pushed onto the browser history stack, so that navigating back goes to the previously selected tab. When initially loading the dashboard navigating back should leave the dashboard.

### How to verify

1. Create a dashboard with tabs.
2. Select different tabs, confirm behavior is as expected.

### Demo

https://github.com/metabase/metabase/assets/37751258/807ebd2d-8a0d-49c0-a381-e2d9e96eaa91

### Checklist
